daemon/src/supervisor.rs

```diff
@@ -0,0 +1,114 @@
+use std::fmt;
+
+use crate::Tasks;
+use async_trait::async_trait;
+use xtra::{Address, Context, Message};
+use xtra_productivity::xtra_productivity;
+
+/// A supervising actor reacts to messages from the actor it is supervising and restarts it based on
```

Comment:
this is amazing. I am really looking forward to using it.
Perhaps someday we could plug-in some "Hypervisor" that could monitor the health of the whole actor system (ie. which are essential to run, which failures are fatal etc.) - some rudimentary way of doing that could be possible even now passing in a custom restart policy (this shows how extendable is this design, nice!).

I guess it would be beneficial for merging the PR to add a few tests of the recovery and (potentially) forwarding into the actor that's being supervised from outside.